### PR TITLE
DO websocket restructuring

### PR DESCRIPTION
**Pull request by joshthoward**

### CODE REVIEW

File: content/durable-objects/api/websockets.md [hunk 1]
```diff
@@ -0,0 +1,261 @@
+---
+title: WebSockets
+pcx_content_type: configuration
+---
+
+# WebSockets
+
+## Background
+
+WebSockets allow you to communicate in real time with your Cloudflare Workers serverless functions. Real time communication combined with Durable Objects as a single-point-of-coordination provides powerful tools for building applications like chat rooms and multiplayer games. For more information beyond the API reference, see [Using WebSockets in Durable Objects](/durable-objects/learning/websockets/).
```
Comment (irvinebroque) [inline content/durable-objects/api/websockets.md]:
```suggestion
WebSockets are long-lived TCP connections that enable bi-directional, real-time communication between client and server.

[Durable Objects](/durable-objects/) support WebSockets — your Durable Object can act as a single point-of-coordination for WebSocket sessions, giving you full control over messages sent to and from clients, allowing you to build applications like chat rooms and multiplayer games.

For more information beyond the API reference, refer to [Using WebSockets in Durable Objects](/durable-objects/learning/websockets/).
```

refs https://github.com/cloudflare/cloudflare-docs/pull/11147

File: content/durable-objects/learning/websockets.md [hunk 2]
```diff
@@ -1,19 +1,46 @@
...
 
-Refer to [Cloudflare Edge Chat Demo](https://github.com/cloudflare/workers-chat-demo) for an example of WebSockets in action within Durable Objects.
\ No newline at end of file
+Code updates will disconnect all WebSockets. If you deploy a new version of a Worker, every Durable Object is restarted. Any connections to old Durable Objects will be disconnected.
+
+{{</Aside>}}
+
+## WebSocket Hibernation
+
+When using a Durable Object on the server side of a WebSocket connection consider using the [Hibernatable WebSockets API](/durable-objects/api/websockets/).
```
Comment (irvinebroque) [inline content/durable-objects/learning/websockets.md]:
```suggestion
When using a Durable Object on the server side of a WebSocket connection, consider using the [Hibernatable WebSockets API](/durable-objects/api/websockets/).
```

**Review by irvinebroque — APPROVED** [review]

File: content/durable-objects/learning/websockets.md [hunk 3]
```diff
@@ -1,19 +1,46 @@
 ---
 pcx_content_type: concept
-title: WebSockets in Durable Objects
+title: Durable Objects with WebSockets
 ---
-# WebSockets in Durable Objects
 
-Durable Objects are a coordinated state tool for [Cloudflare Workers](/workers/), which are often used in parallel with [WebSockets](/workers/runtime-apis/websockets/) to persist state over multiple clients and connections. 
+# Using Durable Objects with WebSockets
```
Comment (Maddy-Cloudflare) [inline content/durable-objects/learning/websockets.md]:
```suggestion
# Use Durable Objects with WebSockets
```

File: content/durable-objects/learning/websockets.md [hunk 4]
```diff
@@ -1,19 +1,46 @@
...
+
+When using a Durable Object on the server side of a WebSocket connection consider using the [Hibernatable WebSockets API](/durable-objects/api/websockets/).
+
+The Hibernatable WebSockets API allows a Durable Object that is not currently running an event handler (such as handling a WebSocket message, HTTP request, or [alarms](/durable-objects/api/alarms-in-durable-objects/)) to be removed from memory while keeping its WebSockets connected ("hibernation").
+
+A Durable Object that hibernates will not incur billable [Duration (GB-sec) charges](/durable-objects/platform/pricing/). For applications with many long-lived Durable Objects and periodic WebSocket messages or events, using the Hibernatable WebSockets API can measurably reduce billable duration.
+
+The Hibernatable WebSockets API includes:
+
+* Cloudflare-specific extensions to the web standard WebSocket API
```
Comment (Maddy-Cloudflare) [inline content/durable-objects/learning/websockets.md]:
```suggestion
* Cloudflare-specific extensions to the web standard WebSocket API.
```

File: content/workers/runtime-apis/websockets.md [hunk 5]
```diff
@@ -7,9 +7,13 @@ title: WebSockets
 
 ## Background
 
-WebSockets allow you to communicate in real time with your Cloudflare Workers serverless functions.
+WebSockets allow you to communicate in real time with your Cloudflare Workers serverless functions. For a complete example, see [Using the WebSockets API](/workers/examples/websockets/).
```
Comment (Maddy-Cloudflare) [inline content/workers/runtime-apis/websockets.md]:
```suggestion
WebSockets allow you to communicate in real time with your Cloudflare Workers serverless functions. For a complete example, refer to [Using the WebSockets API](/workers/examples/websockets/).
```

**Review by Maddy-Cloudflare — APPROVED** [review]

**Pull request merged**